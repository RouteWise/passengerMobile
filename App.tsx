import React, {useState} from 'react';
import {SafeAreaView, View, Text} from 'react-native';
import {COLORS} from './src/constant';
import notifee from '@notifee/react-native';
import {
  BottomDrawer,
  Button,
  SectorSelector,
  Title,
} from './src/components/main';

function App(): JSX.Element {
  const onDisplayNotification = async () => {
    await notifee.requestPermission();

    const channelId = await notifee.createChannel({
      id: 'default',
      name: 'Default Channel',
    });

    // Display a notification
    await notifee
      .displayNotification({
        title: 'Notification Title',
        body: 'Main body content of the notification',
        android: {
          channelId,
          // pressAction is needed if you want the notification to open the app when pressed
          pressAction: {
            id: 'default',
          },
        },
      })
      .then(res => console.log('OK'))
      .catch(err => console.log(err));
  };
  const [open, setOpen] = useState<boolean>(true);

  return (
    <SafeAreaView
      style={{
        flex: 1,
        backgroundColor: COLORS.blue,
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingTop: 50,
      }}>
      <View
        style={{width: '100%', justifyContent: 'center', alignItems: 'center'}}>
        <SectorSelector />
      </View>
      <BottomDrawer open={open}>
        <Title type="medium">Ho≈ügeldiniz</Title>
        <Button
          onPress={() => onDisplayNotification()}
          title="Devam Et"
          type="blueBase"
        />
      </BottomDrawer>
    </SafeAreaView>
  );
}

export default App;
