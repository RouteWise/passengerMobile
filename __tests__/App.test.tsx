/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

import {it, expect} from '@jest/globals';
import {fireEvent, screen, render} from '@testing-library/react-native';

import renderer from 'react-test-renderer';

it('Notification Success Send', () => {
  const tree = render(<App />).toJSON();
  const text = screen.getByTestId('head');
  fireEvent.press(screen.getByTestId('notificationButton'));
});
