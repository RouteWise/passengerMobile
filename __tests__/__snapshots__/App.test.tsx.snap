// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Success Send 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#538DFF",
      "flex": 1,
    }
  }
>
  <Text>
    RouteWise Yolcu
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="notificationButton"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Notifee
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
